core/solver/gmres.cpp

```diff
@@ -159,6 +159,23 @@ void Gmres<ValueType>::apply_impl(const LinOp *b, LinOp *x) const
     auto after_preconditioner =
         matrix::Dense<ValueType>::create_with_config_of(dense_x);
 
+    /* Memory movement summary for average iteration with krylov_dim d:
+     * (5/2d+15/2+12/d)n * values + (1+1/d) * matrix/preconditioner storage
+     * 1x SpMV:                2n * values + storage
+     * 1x Preconditioner:      2n * values + storage
+     * MGS:           (5/2d+5/2)n = sum k=0 to d-1 of (5k+5)n/d
+     *       1x dots           2(k+1)n in iteration k (0-based)
+     *       1x axpys          3(k+1)n in iteration k (0-based)
+     *       1x norm2               n
+     *       1x scal               2n
+     * Restart:         (1+12/d)n  (every dth iteration)
+     *       1x gemm           (d+1)n
+     *       1x Preconditioner     2n * values + storage
+     *       1x axpy               3n
+     *       1x Advanced SpMV      3n * values + storage
+     *       1x norm2               n
+     *       2x scal               2n
```

Comment:
one is scale in initial_2 and the other is add_scale?
add_scale should be 3n memory movement

Reply:
Only a typo, that should be 1x scal, the memory estimate is correct, though